vms/avm/network/gossip.go

```diff
@@ -94,8 +102,11 @@ type gossipMempool struct {
 	txVerifier TxVerifier
 	parser     txs.Parser
 
-	lock  sync.RWMutex
-	bloom *gossip.BloomFilter
+	bloomLock sync.RWMutex
+	bloom     *gossip.BloomFilter
+
+	gossipMempoolLock sync.RWMutex
```

Comment:
Why can't we just use a single lock for this struct?

Reply:
The two are unrelated, and combining them would result in longer locking time.

`gossipMempoolLock` ensures that we have operational atomicity between the gossip.mempool and the txs/mempool instance.

The `bloomLock` ensures that the bloom filter read/write access is being honored.

Reply:
I think this is a pre-mature optimization. Unless we actually see data that indicates that lock contention is something we need to be aware of, I don't think it's worth adding complexity to the code.

Reply:
ok, done.